core/src/main/kotlin/fr/sncf/osrd/envelope_sim_infra/MRSP.kt

```diff
@@ -93,5 +97,89 @@ fun computeMRSP(
                 )
             )
     }
+
+    // Add a limit corresponding to the hardware's maximum operational speed
+    val attrs = listOf(EnvelopeProfile.CONSTANT_SPEED, MRSPEnvelopeBuilder.LimitKind.TRAIN_LIMIT)
+    addSpeedSection(
+        builder,
+        0.meters,
+        pathLength.meters,
+        rsMaxSpeed.metersPerSecond,
+        attrs,
+        speedLimitProperties
+    )
+
+    // Add safety speeds
+    if (safetySpeedRanges != null) {
+        for (range in safetySpeedRanges) {
+            val speed = range.value
+            val attrs =
+                mutableListOf<SelfTypeHolder?>(
+                    EnvelopeProfile.CONSTANT_SPEED,
+                    MRSPEnvelopeBuilder.LimitKind.SPEED_LIMIT,
+                    SpeedLimitSource.SafetyApproachSpeed,
```

Comment:
How about moving `SafetyApproachSpeed` to `LimitKind`?
Semantic looks a bit better, although things are (already) mixed between `LimitKind` and `SpeedLimitSource`.
```suggestion
                    MRSPEnvelopeBuilder.LimitKind.SAFETY_APPROACH_SPEED,
```
Not sure about all this, but we already have the mechanism to have multiple tags (and enforce unicity, which may not be much needed) and it looks like we are making a mess of it (and this is the case prior to this work).